src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_pending_resumption_handler.cc

```diff
@@ -31,252 +31,231 @@
  */
 
 #include "vehicle_info_plugin/vehicle_info_pending_resumption_handler.h"
+#include <boost/range/algorithm/set_algorithm.hpp>
+#include <functional>
 #include "application_manager/event_engine/event_observer.h"
 #include "application_manager/message_helper.h"
 #include "application_manager/resumption/resumption_data_processor.h"
 #include "utils/helpers.h"
-
 namespace vehicle_info_plugin {
+CREATE_LOGGERPTR_GLOBAL(logger_, "VehicleInfoPlugin")
 
-CREATE_LOGGERPTR_GLOBAL(logger_, "VehicleInfoPendingResumptionHandler")
+using hmi_apis::FunctionID::VehicleInfo_SubscribeVehicleData;
 
-VehicleInfoPendingResumptionHandler::VehicleInfoPendingResumptionHandler(
-    application_manager::ApplicationManager& application_manager)
-    : ExtensionPendingResumptionHandler(application_manager) {}
-
-template <class Key, class Value>
-std::set<Key> EnumerateKeys(std::map<Key, Value>& container) {
-  std::set<std::string> keys;
-
-  std::transform(
-      container.begin(),
-      container.end(),
-      std::inserter(keys, keys.end()),
-      [&](const std::pair<std::string, bool>& pair) { return pair.first; });
+uint32_t get_corr_id_from_message(const smart_objects::SmartObject& message) {
+  using namespace application_manager;
+  return message[strings::params][strings::correlation_id].asInt();
+}
 
-  return keys;
+template <class T>
```

Comment:
Please don't forget to get rid of it

Reply:
What is the issue in this method?

Reply:
I also think that this method may be placed in some other file with helper functions, but it is up to you

Reply:
When such method will be required in other files we probably may move it to some common utils